src/toxcore/dht/lan_discovery.rs

```diff
@@ -128,21 +209,25 @@ impl LanDiscoverySender {
 
     /// Run LAN discovery periodically. Result future will never be completed
     /// successfully.
-    pub fn run(mut self) -> IoFuture<()> {
+    pub fn run(mut self) -> impl Future<Item=(), Error=LanDiscoveryError> + Send {
         let interval = Duration::from_secs(LAN_DISCOVERY_INTERVAL);
         let wakeups = Interval::new(Instant::now(), interval);
-        let future = wakeups
-            .map_err(|e| Error::new(ErrorKind::Other, format!("LanDiscovery timer error: {:?}", e)))
+        wakeups
+            .map_err(|e| LanDiscoveryError::from(e))
             .for_each(move |_instant| {
                 trace!("LAN discovery sender wake up");
-                self.send().then(|res| {
-                    if let Err(e) = res {
+                self.send().then(|r| {
+                    if let Err(e) = r {
                         warn!("Failed to send LAN discovery packets: {}", e);
+                        if e.is_inner() {
```

Comment:
No need to check `is_inner` since you check it with `if let` construction.

Reply:
done